src/CovidSim.cpp

```diff
@@ -582,8 +582,9 @@ void ReadParams(char* ParamFile, char* PreParamFile)
 	if (P.DoAdUnits)
 	{
 		char** AdunitNames, * AdunitNamesBuf;
-		if (!(AdunitNames = (char**)malloc(3 * ADUNIT_LOOKUP_SIZE * sizeof(char*)))) ERR_CRITICAL("Unable to allocate temp storage\n");
-		if (!(AdunitNamesBuf = (char*)malloc(3 * ADUNIT_LOOKUP_SIZE * 360 * sizeof(char)))) ERR_CRITICAL("Unable to allocate temp storage\n");
+		if (!(AdunitNames = (char**)malloc(3 * ADUNIT_LOOKUP_SIZE * sizeof(char*))) ||
+			!(AdunitNamesBuf = (char*)malloc(3 * ADUNIT_LOOKUP_SIZE * 360 * sizeof(char))))
+			ERR_CRITICAL("Unable to allocate temp storage\n");
```

Comment:
Hi @zdroid, thanks for this. I'm afraid I agree with @matt-gretton-dann though, I think if anything we'd prefer the ERR_CRITICAL calls to be more specific rather than less. So would be really helpful to have e.g. these lines broken up to display "Unable to allocate temp storage - AdunitNames\n" and "Unable to allocate temp storage - AdunitNamesBuf\n"